### ISSUE TITLE
Issue running pipeline

### ISSUE THREAD

**Issue by tvanhateren** [mhardcastle/ddf-pipeline#166]
Trying to run the pipeline on data from the lofar archive.
This time using target: P23, observation id L232875.
Using 20 of the available ~240 sub-bands.

The data has been run through the prefactor target parset, with the cal_values from the calibrator.

Problem is, now all the flux levels show as zero in ddf-pipeline.

[DDF-image_dirin_SSD.log](https://github.com/mhardcastle/ddf-pipeline/files/2977662/DDF-image_dirin_SSD.log)

**Comment by mhardcastle** [comment]
This is usually a sign that your CORRECTED_DATA column is all zeros, which it will be if it was uninitialized going into the pipeline. Perhaps your calibrated data are in DATA? Set [data] colname=DATA in the config file to work there.

Note that the pipeline isn't designed to work with only two 10-SB MS as input. I would turn bootstrap off, for a start, but you may also run into issues with the self-calibration.

Also... I presume you know that the surveys KSP has processed that field, and the images are available for download as part of DR1? (I realise the most likely thing is that you're using it as test data...)

**Comment by tvanhateren** [comment]
Indeed, the measurement sets do not have a column CORRECTED_DATA. So most likely using the DATA column.
And yes, I am using this dataset just to test the the pipeline on local infrastructure. I don't need the images themselves.

I am now running the pipeline with new settings, and it does now find non-zero flux values. So it looks like this problem is solved.

**Comment by tvanhateren** [comment]
With the correct column name the pipeline progresses a bit further but stops at a shape mismatch:
[DDF-image_phase1.log](https://github.com/mhardcastle/ddf-pipeline/files/2984229/DDF-image_phase1.log)

**Comment by cyriltasse** [comment]
try to `rm -rf *.ddfcache` and run again

**Comment by tvanhateren** [comment]
That run was already done with a clean cache, not a rerun on the same cache

**Comment by mhardcastle** [comment]
I think this may be a problem that's been fixed in current (but private) versions of DDF...

**Comment by tvanhateren** [comment]
Any chance of getting a copy of that private version?
Or alternatively is there a way to work around the problem?

**Comment by mhardcastle** [comment]
Best to take this to e-mail I think. Can you get in contact with me that way and we'll discuss?

**Comment by cyriltasse** [comment]
The log says things about cached parameters, so it wasn't rerun with a clean cache...?

```
2019-03-18 20:11:04:  - 20:11:04 - CacheManager              | cache hash /local/thn215/cache/mslist.txt.ddfcache/Dirty.hash does not match, will re-make
2019-03-18 20:11:04:  - 20:11:04 - CacheManager              |   differences in parameters (Param: this vs cached): (Weight.ColName: IMAGING_WEIGHT vs None) & (InitDicoModel: <type 'str'> vs <type 'NoneType'>) & (DDESolutions.DDSols: killms_p1 vs ) & (DDESolutions.GlobalNorm: BLBased vs None) & (DDESolutions.DDModeGrid: P vs AP) & (DDESolutions.DDModeDeGrid: P vs AP)
2019-03-18 20:11:07:  - 20:11:07 - CacheManager              | cache hash /local/thn215/cache/mslist.txt.ddfcache/PSF.hash does not match, will re-make
2019-03-18 20:11:07:  - 20:11:07 - CacheManager              |   differences in parameters (Param: this vs cached): (Weight.ColName: IMAGING_WEIGHT vs None) & (InitDicoModel: <type 'str'> vs <type 'NoneType'>) & (DDESolutions.DDSols: killms_p1 vs ) & (DDESolutions.GlobalNorm: BLBased vs None) & (DDESolutions.DDModeGrid: P vs AP) & (DDESolutions.DDModeDeGrid: P vs AP)

```

**Comment by tvanhateren** [comment]
This cache is created during the run. Before calling ddf the cache and work directories are completely empty.

In the mean time I have calibrated the same data with factor, so the input data looks to be in good shape.

**Comment by cyriltasse** [comment]
That's not what the log says...?

**Comment by tvanhateren** [comment]
I think you mean I should run the pipeline till it crashes, then delete the cache and run it again.

**Comment by cyriltasse** [comment]
yes! :)

**Comment by tvanhateren** [comment]
Next hickup at https://github.com/mhardcastle/ddf-pipeline/blob/0587ef12f74dc541074111920526a83d3929e247/utils/auxcodes.py#L214-L218
The MakeMask.py command stores the output as `rms.fits`, but the code is looking for input at `rms.fits.fits`
For right now I just made a symlink and continue.

**Comment by tvanhateren** [comment]
Issue: The Pan-STARRS API has been changed with data release 2, so I have changed this request
https://github.com/mhardcastle/ddf-pipeline/blob/0587ef12f74dc541074111920526a83d3929e247/utils/get_cat.py#L44
to
```python
r = requests.post('https://catalogs.mast.stsci.edu/api/v0.1/panstarrs/dr1/mean.votable', data = {'ra':p[0],'dec':p[1],'radius':CSIZE,'pagesize':100000,'nDetections.gte':"5",'action':'Search','columns':'[objid,ramean,decmean]'},timeout=300)
```

This looks like it worked correctly, we can find matching sources from the target field in the combined fits image from Pan-STARSS, however after that we get: 
```
Running: offsets.py /home/thn215/DDF/232875.cfg
Determining astrometric offsets with method panstarrs in mode normal
Merging downloaded catalogues
Running PyBDSM on LOFAR image, please wait...
--> Opened 'image_full_ampphase1m.int.restored.fits'
Image size .............................. : (19845, 19845) pixels
Number of channels ...................... : 1
Number of Stokes parameters ............. : 1
Beam shape (major, minor, pos angle) .... : (1.66667e-03, 1.66667e-03, 90.0) degrees
Frequency of image ...................... : 1415.000 MHz
Number of blank pixels .................. : 0 (0.0%)
 - 11:29:39 - ClearSHM                  | Clear shared memory
 - 11:29:39 - ClearSHM                  | Clear Semaphores
 - 11:29:39 - ClearSHM                  | Clear shared dictionaries
FAILED to run offsets.py /home/thn215/DDF/232875.cfg: return value is -11
Traceback (most recent call last):
  File "/opt/lofar/ddf-pipeline/scripts/pipeline.py", line 661, in <module>
    run('offsets.py '+' '.join(sys.argv[1:]),log=None)
  File "/opt/lofar/ddf-pipeline/utils/auxcodes.py", line 45, in run
    die('FAILED to run '+s+': return value is '+str(retval))
  File "/opt/lofar/ddf-pipeline/utils/auxcodes.py", line 27, in die
    raise Exception(s)
Exception: FAILED to run offsets.py /home/thn215/DDF/232875.cfg: return value is -11 
```

offsets.py crashes with a segfault

I will do some more exploring as to which code is segfaulting

**Comment by mhardcastle** [comment]
Most of these issues are already fixed in pipeline versions 2.0 and up.
